src/main/java/at/petrak/hexcasting/common/blocks/circles/BlockEntitySlate.java

```diff
@@ -21,6 +25,19 @@ public BlockEntitySlate(BlockPos pos, BlockState state) {
         super(HexBlockEntities.SLATE_TILE, pos, state);
     }
 
+    @Override
+    protected void collectImplicitComponents(DataComponentMap.Builder components) {
+        super.collectImplicitComponents(components);
+        if (this.pattern != null) {
+            var tag = new CompoundTag();
+            tag.putString("id", HexAPI.modLoc("slate").toString());
+            tag.put(TAG_PATTERN, HexPattern.CODEC
+                    .encodeStart(NbtOps.INSTANCE, this.pattern)
+                    .getOrThrow());
+            components.set(DataComponents.BLOCK_ENTITY_DATA, CustomData.of(tag));
```

Comment:
Is there any reason to use `BlockEntityData` instead of custom data component here?
You can save it to `ItemStack` via this method, and load it from `ItemStack` via `applyImplicitComponents`. I think it would be nicer

Reply:
Oh I just used `BlockEntityData` because that looks like what was intended to be used in the prior `ItemSlate` implementation. Are you saying that the `collectImplicitComponents` implementation should update the `BlockEntityData` to add the `TAG_PATTERN` like is done here for `ItemSlate` https://github.com/slava110/NeoHexCasting/blob/022e83e229461e12accb255d77a46dd025559637/src/main/java/at/petrak/hexcasting/common/items/storage/ItemSlate.java#L119-L127 or are you suggesting to completely circumvent `BlockEntityData` altogether and make a different data component with `TAG_PATTERN` on it for both the `ItemSlate` and `BlockEntitySlate`?

Reply:
Actually I think I saw what you meant to do here, went ahead and fixed it! Definitely looks a lot cleaner.